tests_tf/test_attacks.py

```diff
@@ -865,7 +867,13 @@ def test_multiple_initial_random_step(self):
             I = (orig_labs == new_labs_multi)
             new_labs_multi[I] = new_labs[I]
 
-        self.assertTrue(np.mean(orig_labs == new_labs_multi) < 0.1)
+        self.assertTrue(np.mean(orig_labs == new_labs_multi) < 0.5)
+
+
+class TestProjectedGradientDescent(TestMadryEtAl):
```

Comment:
Could you add a separate test for the MadryEtAl class and the BasicIterativeMethod class? I found that some of the code broke for one but not the other when trying to consolidate the code earlier so I'm worried that changes down the road might introduce bugs we don't notice if we don't have a simple test for both classes in addition to the ProjectedGradientDescent class.

Reply:
Done